### ISSUE TITLE
sos report doesn't enable plugins specified on command line

### ISSUE THREAD

**Issue by rmetrich** [sosreport/sos#2264]
When a plugin is disabled in `/etc/sos.conf` through using `skip-plugins` for example, it's not possible to enable the plugin anyway using the command line `sos report -e <plugin>`.
I think this is a bug, command line should take precedence over configuration file.

**Comment by bmr-cymru** [comment]
The bug is that the behaviour is not documented - I don't think it's as straightforward as saying "command line takes precedence", thinking back to when the config file disable option was first introduced one of the use-cases was to allow administrators to disable a problematic plugin across the board.

The fact that the configuration file has allowed you to prevent plugins from being used on the command line is long standing behaviour - if the consensus is to change that it's fine, but this time we should record the decision and make sure it is unambiguously documented for users.

**Comment by rmetrich** [comment]
Well IMHO command line should take precedence, exactly as all programs do. Example sshd reads /etc/ssh/sshd_config but allows an override through -p flag.
Here, in my use case, the customer disables plugins because they are too slow / collecting too much data by default, for example when sosreport executes for abrt reports.

But when Red Hat support requires the plugins output anyway to investigate a specific output, the customer should be able to just use `-e <plugin>` option to override his default settings.
Currently he needs to either modify /etc/sos.conf or use an alternate dummy sos.conf, e.g. `sosreport --config-file /dev/null`

**Comment by bmr-cymru** [comment]
> Well IMHO command line should take precedence, exactly as all programs do.

That's common behaviour but clearly not universal: I can think of a half dozen examples just from the top of my head where a configuration file has precedence over command line options (particularly where the intent is to allow something to be completely disabled, which was the original intent here).

> Here, in my use case, the customer disables plugins because they are too slow / collecting too much data by default, for example when sosreport executes for abrt reports.

It's better to fix this by complaining to `abrt` - they historically hard-code a complicated `sosreport` command line that includes explicit lists of plugins to use. Disabling things via the config file may then cause abrt runs of `sos` to fail completely if there are no remaining valid plugins.

It's probably better to just disable `abrt-sosreport-plugin` if you don't want the reports, or to modify `/etc/libreport/events.d/sosreport_event.conf` to override the default settings provided by the Abrt project:

```
# Example: if you want to save sosreport immediately at the moment of a crash:
# (alternatively, you can add similar command to EVENT=analyze_foo's,
# if you would rather perform this collection later):
EVENT=post-create remote!=1
        nice sosreport --tmp-dir "$DUMP_DIR" --batch \
                --only=anaconda --only=boot --only=devicemapper \
                --only=filesys --only=hardware --only=kernel --only=libraries \
                --only=memory --only=networking --only=nfsserver --only=pam \
                --only=process --only=rpm -k rpm.rpmva=off --only=ssh \
                --only=startup --only=yum --only=date --only=host --only=x11 \
                --only=cups --only=logs --only=grub2 --only=cron --only=pci \
                --only=auditd --only=selinux --only=lvm2 --only=sar \
                --only=processor \
                >sosreport.log 2>&1 \
        && {
                rm sosreport.log
                rm sosreport*.md5
                mv sosreport*.tar.bz2 sosreport.tar.bz2
                mv sosreport*.tar.xz sosreport.tar.xz
                exit 0
        } 2>/dev/null
        # Error in sosreport run. Let user see the problem.
        echo "sosreport run failed with exit code $?, log follows:"
        # sosreport prints many useless empty lines, nuke them:
        # it looks awful in syslog otherwise.
        cat sosreport.log | sed 's/  *$//' | grep -v '^$'
        rm sosreport.log
        exit 1
```

Anyway, I'm not maintaining `sos` these days so the decision rests with the current maintainer and contributors: the most important thing is that whatever behaviour is decided on it is consistent and properly documented.

**Comment by TurboTurtle** [comment]
Yeah, this has nothing to do with command line precedence over the config file - this is priority of `enable` (or even `only`) vs `skip`. If you try to run `sos report -o lvm2,host -n host` you'll find you get the same behavior - `host` is skipped.

My personal opinion is that this is preferable to the reverse - for exactly the reason @bmr-cymru highlights; it places ultimate control of how sos behaves in a system administrator's hands when you have conflicts. 

I am open to being persuaded otherwise, however.

**Comment by TurboTurtle** [comment]
> Currently he needs to either modify /etc/sos.conf or use an alternate dummy sos.conf, e.g. sosreport --config-file /dev/null

The other option is override the `skip-plugins` option on the command line, for instance "skipping" an already disabled plugin (e.g. `zfs` on almost any RHEL system).

We could perhaps add a check for `none` being passed to `skip-plugins` to clear any plugins skipped by the config file, but I am wary of that.
